api/bibliotheca.py

```diff
@@ -1283,32 +1311,62 @@ def _process_metadata(
 
-    """Register CirculationEvents for Bibliotheca titles.
+class BibliothecaTimelineMonitor(CollectionMonitor, TimelineMonitor):
+    """Common superclass for our two TimelineMonitors."""
+    PROTOCOL = ExternalIntegration.BIBLIOTHECA
+    LOG_DATE_FORMAT = '%Y-%m-%dT%H:%M:%S'
 
-    Most of the time we will just be finding out that someone checked
-    in or checked out a copy of a book we already knew about.
+    def __init__(self, _db, collection, api_class=BibliothecaAPI,
+                 analytics=None):
+        """Initializer.
 
-    But when a new book comes on the scene, this is where we first
-    find out about it. When this happens, we create a LicensePool and
-    immediately ensure that we get coverage from the
-    BibliothecaBibliographicCoverageProvider.
+        :param _db: Database session object.
 
-    But getting up-to-date circulation data for that new book requires
-    either that we process further events, or that we encounter it in
-    the BibliothecaCirculationSweep.
-    """
+        :param collection: Collection for which this monitor operates.
 
-    SERVICE_NAME = "Bibliotheca Event Monitor"
-    DEFAULT_START_TIME = timedelta(days=365*3)
-    PROTOCOL = ExternalIntegration.BIBLIOTHECA
-    LOG_DATE_FORMAT = '%Y-%m-%dT%H:%M:%S'
+        :param api_class: API class or an instance thereof for this monitor.
+        :param api_class: Union[Type[BibliothecaAPI], BibliothecaAPI]
+
+        :param analytics: An optional Analytics object.
+        :type analytics: Optional[Analytics]
+        """
+        self.analytics = analytics or Analytics(_db)
+        super(BibliothecaTimelineMonitor, self).__init__(_db, collection)
+        if isinstance(api_class, BibliothecaAPI):
+            # We were given an actual API object. Just use it.
+            self.api = api_class
+        else:
+            self.api = api_class(_db, collection)
+        self.replacement_policy = BibliothecaAPI.replacement_policy(
+            _db, self.analytics
+        )
+        self.bibliographic_coverage_provider = BibliothecaBibliographicCoverageProvider(
+            collection, self.api, replacement_policy=self.replacement_policy
+        )
+
+
+class BibliothecaPurchaseMonitor(BibliothecaTimelineMonitor):
+    """Track purchases of licenses from Bibliotheca.
+
+    Most TimelineMonitors monitor the timeline starting at whatever
+    time they're first run. But it's crucial that this monitor start
+    at or before the first day on which a book was added to this
+    collection, even if that date was years in the past. That's
+    because this monitor may be the only time we hear about a
+    particular book.
+
+    Because of this, this monitor has a very old DEFAULT_START_TIME
+    and special capabilities for customizing the start_time to go back
+    even further.
+    """
+    SERVICE_NAME = "Bibliotheca Purchase Monitor"
+    DEFAULT_START_TIME = datetime_utc(2014, 1, 1)
```

Comment:
DEFAULT_START_TIME has moved from the event monitor, and changed from a timedelta (3 years in the past) to a constant date (1/1/2014).